### DISCUSSION TITLE
Let user choose about attachment handling

### DISCUSSION THREAD

**Discussion by cm-schl** [the-djmaze/snappymail#1173] [Ideas]
Hi @the-djmaze,
I searched inside the issues but did not find if this is even possible. As there were some issues about the .eml attachments I made some tests and would like to discuss something.
Is there really some eml viewer inside of SnappyMail that will try to show the content of the .eml file? Or is SnappyMail only passing the correct mime type `message/rfc822` to the browser and therefore the browser is handling the visualization of the .eml?

The problem that I see actually is, that some mails are not shown correctly when the user clicks on the .eml attachment and there seams no easy possibility to open the .eml with a local program (on computer, not mobile). The only workaround that I found is to download the attachment as zip and open it locally afterwards...

From the mobile first perspective to have a .eml viewer is perfectly fine - but maybe there should be an option to open the mail with a local program. What do you say about this idea?

**Comment by the-djmaze** [comment]
The "view in separate window" code is here:
https://github.com/the-djmaze/snappymail/blob/master/dev/Html/PreviewMessage.html
https://github.com/the-djmaze/snappymail/blob/ed31ed206d47c72eb1e7b6e110d983af84b618cb/dev/Model/Message.js#L339-L364

Difference between RainLoop and SnappyMail is that SnappyMail can parse EML into a `MessageModel` (RainLoop can not) using https://github.com/the-djmaze/snappymail/tree/master/dev/Mime
Then SnappyMail uses the same code as RainLoop to view it using `MessageModel.viewPopupMessage()`

To see attachments, the code needs a complete rewrite with enhancements to handle attachments.

**Reply by cm-schl** [reply]
Ok, thanks for the information 👍.
I'm fine if SnappyMail can open the .eml directly in the browser, this is nice. But that the user can't download an .eml (besides the download as zip workaround) is a bit irritating for users. If the user clicks on the name of the file, the eml viewer opens. If the users clicks on the eye icon, the raw content opens...

Would be adding a download link inside the eml viewer solve the problem? So the code could remain the same in sense of opening the eml viewer, but the user has always the choice to download the original eml. This would also mitigate the situation in https://github.com/the-djmaze/snappymail/issues/1166 a bit - nevertheless this isn't a real solution to the question.
